### PR TITLE
Target 3.11-rc.1 in `nightly` python build

### PR DESCRIPTION
**Pull request by scbedd**

### CODE REVIEW

**Review by mccoyp — APPROVED** [review]
Thank you for beating me to this! Should we also add a `sudo apt-get install build-essential` to the Linux pipeline, or merge this first and see if it turns out to be necessary?

**Comment by scbedd** [comment]
I see where you're going with that. I'll add it and merge this.

File: eng/pipelines/templates/jobs/tests-nightly-python.yml
Comment (scbedd) [inline eng/pipelines/templates/jobs/tests-nightly-python.yml:64]:
```suggestion
```

File: eng/pipelines/templates/jobs/tests-nightly-python.yml
Comment (scbedd) [inline eng/pipelines/templates/jobs/tests-nightly-python.yml:34]: We will disable compatibility by passing filter type `"None"` to this parameter when it is available after another PR merges.

**Review by mccoyp — APPROVED** [review]

**Pull request merged**
